tests/test_plot_manager.py

```diff
@@ -0,0 +1,177 @@
+        tests/common/single-model-ckpt
+        """
+        plot_manager = PlotManager(
+            config=self._single_model_config,
+            result_manager=self._single_model_result_manager)
+
+        plot_manager.create_summary_plots()
+
+        # Uncomment these lines to create a new output dump to compare against
+        # file = open(f"{ROOT_DIR}/single-model-ckpt/plot_manager.pkl", "wb")
+        # pickle.dump(plot_manager._simple_plots, file)
+        # file.close()
+
+        file = open(f"{ROOT_DIR}/single-model-ckpt/plot_manager.pkl", "rb")
+        golden_plot_manager_simple_plots = pickle.load(file)
+        file.close()
+
+        golden = golden_plot_manager_simple_plots['add_sub'][
+            'throughput_v_latency']
+        plot = plot_manager._simple_plots['add_sub']['throughput_v_latency']
+        self._compare_plot_data(golden, plot)
+
+        golden = golden_plot_manager_simple_plots['add_sub'][
+            'gpu_mem_v_latency']
+        plot = plot_manager._simple_plots['add_sub']['gpu_mem_v_latency']
+        self._compare_plot_data(golden, plot)
+
+    def test_multi_model_summary_plots_against_golden(self):
+        """
+        Match the summary plots against the golden versions in
+        tests/common/multi-model-ckpt
+        """
+        plot_manager = PlotManager(
+            config=self._multi_model_config,
+            result_manager=self._multi_model_result_manager)
+
+        plot_manager.create_summary_plots()
+
+        # Uncomment these lines to create a new output dump to compare against
+        # file = open(f"{ROOT_DIR}/multi-model-ckpt/plot_manager.pkl", "wb")
+        # pickle.dump(plot_manager._simple_plots, file)
+        # file.close()
+
+        file = open(f"{ROOT_DIR}/multi-model-ckpt/plot_manager.pkl", "rb")
+        golden_plot_manager_simple_plots = pickle.load(file)
+        file.close()
+
+        golden = golden_plot_manager_simple_plots[
+            'resnet50_libtorch,vgg19_libtorch']['throughput_v_latency']
+        plot = plot_manager._simple_plots['resnet50_libtorch,vgg19_libtorch'][
+            'throughput_v_latency']
+        self._compare_plot_data(golden, plot)
+
+        golden = golden_plot_manager_simple_plots[
+            'resnet50_libtorch,vgg19_libtorch']['gpu_mem_v_latency']
+        plot = plot_manager._simple_plots['resnet50_libtorch,vgg19_libtorch'][
+            'gpu_mem_v_latency']
+        self._compare_plot_data(golden, plot)
+
+    def _compare_plot_data(self, golden, plot):
+        self.assertEqual(golden._name, plot._name)
+        self.assertEqual(golden._title, plot._title)
+        self.assertEqual(golden._x_axis, plot._x_axis)
+        self.assertEqual(golden._y_axis, plot._y_axis)
+        self.assertEqual(golden._monotonic, plot._monotonic)
+        self.assertEqual(golden._data, plot._data)
+
+        # These are generated by pyplot and will vary from run to run
+        #self.assertEqual(golden._fig, plot._fig)
```

Comment:
if these aren't comparable due to randomness, remove the commented out code from testing.

Reply:
This is documentation so that it doesn't look like I omitted comparing part of the classes data.

Reply:
oic. cool cool cool